plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiReadOptimizedDirectoryLister.java

```diff
@@ -92,33 +67,29 @@ public List<HudiPartitionInfo> getPartitionsToScan()
                         hiveTable,
                         hiveMetastore))
                 .collect(Collectors.toList());
-
-        return allPartitionInfoList.stream()
-                .filter(partitionInfo -> partitionInfo.getHivePartitionKeys().isEmpty() || partitionInfo.doesMatchPredicates())
```

Comment:
Do we still run `partitionInfo.doesMatchPredicates()` someplace? This might help reducing the external loading of `partitionInfo.getHivePartitionKeys()` which calls directly to Hive.

Reply:
Good catch! We should call it in `HudiPartitionInfoLoader`.